velox/common/memory/MallocAllocator.cpp

```diff
@@ -239,36 +234,20 @@ int64_t MallocAllocator::freeNonContiguous(Allocation& allocation) {
   if (allocation.empty()) {
     return 0;
   }
-  MachinePageCount numFreed = 0;
+  MachinePageCount freedPages{0};
   for (int32_t i = 0; i < allocation.numRuns(); ++i) {
     Allocation::PageRun run = allocation.runAt(i);
     void* ptr = run.data();
-    int64_t numPages;
-    {
-      std::lock_guard<std::mutex> l(nonContiguousMallocsMutex_);
-      const auto it = nonContiguousMallocs_.find(ptr);
-      VELOX_CHECK(
-          it != nonContiguousMallocs_.end(), "Bad free page pointer: {}", ptr);
-      numPages = static_cast<int64_t>(it->second);
-      nonContiguousMallocs_.erase(it);
-    }
-    numFreed += numPages;
+    const int64_t numPages = run.numPages();
```

Comment:
do we mention this explicitly anywhere?
`ensuring each page run in non-contiguous allocation is a single buffer allocated
from jemalloc to avoid the lock.`

Would be good to maybe mention this in the Malloc allocator and also mention how we ensure this by making sure a PageRun has a limit on the max number of pages.